lib/fetch-package-metadata.js

```diff
@@ -304,16 +300,19 @@ function untarStream (tarball, cb) {
       er.code = 'EUNTAR'
       cb(er)
     })
-    tounpipe.push(untar)
+    tounpipe.push([stream, untar])
     stream = untar
     addClose()
   }
 
   function addClose () {
     stream.close = function () {
-      tounpipe.forEach(dispose)
+      tounpipe.forEach(function (streams) {
+        unpipe(streams[0],streams[1])
```

Comment:
afaict, `unpipe` only takes one argument, and it unpipes _all_ piped streams. See https://www.npmjs.com/package/unpipe (and source: https://github.com/stream-utils/unpipe/blob/master/index.js#L40)